fips/usr/lib/systemd/system/snap-initramfs-mounts.service.d/fips-img.conf

```diff
@@ -0,0 +1,15 @@
+[Unit]
+After=systemd-modules-load.service
+
+# This is for kernels that are not booted from an EFI environment. This probably
+# means a FIT image on arm64.
+ConditionFirmware=!uefi
+
+[Service]
+
+# Ensure tcrypt was loaded
+ExecStartPre=modprobe tcrypt
+# Then allow snap-bootstrap to mount things
+# Self-test verify FIPS computed hmac
+ExecStartPost=kcapi-hasher -n sha512hmac --check /run/mnt/kernel/.kernel.img.hmac
```

Comment:
Note, whilst it is a convention to name hmac checksum file as .$ORIG_FILE.hmac, we don't have to, as it itself encodes the filename to check.

Thus if we ever desire to support alternative filenames, we could conceivably change the .hmac file name to .kernel.hmac, encode name with extension inside it, and have a universal check, if we still use or desire such a check in the future. For example, it is likely that UEFI boot will mvoe away from this check in favour of kexec checks.